tap/tcp_reader.go

```diff
@@ -81,14 +79,23 @@ func (reader *tcpReader) sendMsgIfNotClosed(msg api.TcpReaderDataMsg) {
 	reader.Unlock()
 }
 
+func (reader *tcpReader) isProtocolIdentified() bool {
+	return reader.parent.protocol != nil
+}
+
 func (reader *tcpReader) Read(p []byte) (int, error) {
+	if reader.exhaustBuffer {
+		reader.data = reader.buffer
+		reader.exhaustBuffer = false
+	}
```

Comment:
```suggestion
func (reader *tcpReader) rewind() {
	reader.data = reader.buffer
}

func (reader *tcpReader) Read(p []byte) (int, error) {
```

No need for `reader.exhaustBuffer`.
Call `reader.rewind` where you would set `reader.exhaustBuffer = true`.

Reply:
@nimrod-up9 done with https://github.com/up9inc/mizu/commit/8c8eb20ad6ea7324c2fa518fc836fa0ab31f4b6d